shared/fs/row/uploading.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+import * as React from 'react'
+import * as Types from '../../constants/types/fs'
+import {globalColors, isMobile} from '../../styles'
+import rowStyles from './styles'
+import {Box, Icon, Meta, Text, Divider} from '../../common-adapters'
+import PathItemIcon from '../common/path-item-icon'
+
+type UploadingProps = {
+  name: string,
+  itemStyles: Types.ItemStyles,
+}
+
+const Uploading = (props: UploadingProps) => (
+  <Box>
+    <Box style={rowStyles.row}>
+      <PathItemIcon spec={props.itemStyles.iconSpec} style={rowStyles.pathItemIcon_30} />
+      <Box style={stylesIconBadge}>
+        <Icon type="icon-addon-file-uploading" />
+      </Box>
+      <Box key="main" style={rowStyles.itemBox}>
+        <Text
+          type={props.itemStyles.textType}
+          style={rowStyles.rowText_30(props.itemStyles.textColor)}
+          lineClamp={isMobile ? 1 : undefined}
+        >
+          {props.name}
+        </Text>
+        <Meta title="Encrypting & Uploading" backgroundColor={globalColors.blue} />
+      </Box>
+    </Box>
+    <Divider style={rowStyles.divider} />
+  </Box>
+)
+
+const xOffset = -28
+const yOffset = 20
+const stylesIconBadge = {
+  marginLeft: xOffset,
+  marginRight: -xOffset,
+  marginTop: yOffset,
+  width: 0,
+  zIndex: 100,
```

Comment:
is this really necessary

Reply:
Seems like so. Somehow it's affected by opacity of the thing under it:

<img width="208" alt="screen shot 2018-06-08 at 11 45 48 am" src="https://user-images.githubusercontent.com/255797/41175344-9d4036da-6b11-11e8-87b6-f1ff2f736c41.png">

Is there an alternative here?

Reply:
just put it later in the dom

Reply:
I placed this as the last in the containing Box (and changed to row-reverse because it needs the fix-sized PathItemIcon to compute offset), and it didn't work.

Reply:
(still affected by opacity of underlying component)

Reply:
lemme look

Reply:
Thansk!

Reply:
ok so this is because a new stacking context is created since the icon has an opacity applied to it. so that forces you to do something like this which is kind of annoying. so we can keep it for now. if you happend to get a semi transparent image from cecile it actually would have worked

Reply:
Hmm that's beyond my knowledge to understand. A semi transparent image meaning there's at least one pixel in the PNG that has an effective alpha value? It's still nice to not dependent on PNG having that though I guess.

Reply:
No I mean you have an opacity value applied through css which causes this behavior. If the transparency want handled this way it'd stack as expected

Reply:
Ah I see. Makes sense.